godal.cpp

```diff
@@ -1075,7 +1095,7 @@ OGRFeatureH godalLayerNewFeature(cctx *ctx, OGRLayerH layer, OGRGeometryH geom)
 	if (hFeature!=nullptr && geom!=nullptr) {
 		oe = OGR_F_SetGeometry(hFeature,geom);
 		if (oe != OGRERR_NONE) {
-			oe = OGR_L_SetFeature(layer,hFeature);
+			oe = OGR_L_CreateFeature(layer,hFeature);
```

Comment:
Do you have an explanation as to why the previous code seemed to be working correctly here? Or do you have a testcase that failed here?

Reply:
In fact, SetFeature or CreateFeature will give same result.
Just see that real problem is at line 1097, `oe != OGRERR_NONE` must be replaced by `oe == OGRERR_NONE`.
I will fix it and a unit test to check that feature is properly added (currently not tested).